ide/libs.git/src/org/netbeans/libs/git/jgit/JGitSshSessionFactory.java

```diff
@@ -167,9 +177,25 @@ protected Session createSession (Host hc, String user, String host, int port, FS
     private boolean setupJSchIdentityRepository (JSch jsch, String identityFile, boolean preferAgent) throws JSchException {
         boolean agentUsed = false;
         if (preferAgent) {
-            Connector con = ConnectorFactory.getInstance().createConnector(ConnectorFactory.ConnectorKind.ANY);
-            if (con != null) {
-                IdentityRepository irepo = new IdentityRepositoryImpl(con);
+            AgentConnector agentConnector = null;
+            try {
+                 agentConnector = new SSHAgentConnector();
+                 if(! agentConnector.isAvailable()) {
+                     agentConnector = null;
+                 }
+            } catch (Throwable ex) {
+                LOG.log(Level.FINE, null, ex);
+            }
+            try {
+                agentConnector = new PageantConnector();
```

Comment:
Why create a new "PageantConnector()" on line 190 if we've already created a SSHAgentConnector on line 182 above?

Reply:
There are two possible outcomes of the creation of the SSHAgentConnector:

- instantiation of `SSHAgentConnector` could fail because `junixsocket` can't be instatiated
- the class might be loadable, but there is no agent running

On window this is quite possible. In my work setup I have pagent running and an ssh-agent -> pagent bridge in cygwin, but NetBeans uses PAgent directly.

Reply:
What I was meaning is that you may want an `if (agentconnector != null) {}` wrapping the second try-catch, or not?